interop/tapir/src/main/scala/caliban/interop/tapir/TapirAdapter.scala

```diff
@@ -254,77 +276,22 @@ object TapirAdapter {
-                                     val query         = fields.get("query").collect { case StringValue(v) => v }
-                                     val operationName = fields.get("operationName").collect { case StringValue(v) => v }
-                                     val variables     = fields.get("variables").collect { case InputValue.ObjectValue(v) => v }
-                                     val extensions    = fields.get("extensions").collect { case InputValue.ObjectValue(v) => v }
-                                     GraphQLRequest(query, operationName, variables, extensions)
-                                   }
-                                   request match {
-                                     case Some(req) =>
-                                       val stream = generateGraphQLResponse(
-                                         req,
-                                         id.getOrElse(""),
-                                         interpreter,
-                                         skipValidation,
-                                         enableIntrospection,
-                                         queryExecution,
-                                         subscriptions
-                                       )
-                                       webSocketHooks.onMessage
-                                         .map(_.transform(stream))
-                                         .getOrElse(stream)
-                                         .mapM(output.offer)
-                                         .runDrain
-                                         .catchAll(e => output.offer(makeError(id, e)))
-                                         .fork
-                                         .interruptible
-                                         .unit
-
-                                     case None => output.offer(connectionError)
-                                   }
-                                 case GraphQLWSInput("stop", id, _)                  =>
-                                   removeSubscription(id, subscriptions)
-                                 case GraphQLWSInput("connection_terminate", _, _)   =>
-                                   ZIO.interrupt
-                               }.runDrain.interruptible
-                                 .catchAll(_ => output.offer(connectionError))
-                                 .ensuring(subscriptions.get.flatMap(m => ZIO.foreach(m.values)(_.succeed(()))))
-                                 .provide(env)
-                                 .forkDaemon
-                             )(_.interrupt) *> ZStream.fromQueueWithShutdown(output)
-                         }
-      } yield pipe
-
+  ): ServerEndpoint[ZioWebSockets, RIO[R, *]] =
     makeWebSocketEndpoint.serverLogic[RIO[R, *]](serverRequest =>
-      requestInterceptor(serverRequest)(io).catchAll(ZIO.left(_))
+      requestInterceptor(serverRequest._1)(
```

Comment:
How about `{ case (something, somethingElse) =>` instead of `(serverRequest =>` so that it reads more easily?

Reply:
👍